### DISCUSSION TITLE
I2S Slave Mode

### DISCUSSION THREAD

**Discussion by AntonioBanditoz** [pschatzmann/ESP32-A2DP#449] [Q&A] [answered]
Hi everyone, how can I configure the module to accept external bclk and lrck? I need to send i2s signal to adau1701. The bclk lrck clocking will be from adau1701. Thanks

**Comment by pschatzmann** [comment, marked as answer]
This is a really strange question: and you already gave the answer yourself: Instead of master mode you would define slave mode. Just read the Readme again: there it is described how you you can define your custom i2s mode.

Further information can be found in the official ESP32 documentation: https://docs.espressif.com/projects/esp-idf/en/release-v4.4/esp32/api-reference/peripherals/i2s.html

**Reply by AntonioBanditoz** [reply]
I understand, but it didn't work for me, so I thought I'd ask, I did it like this:

.mode = (i2s_mode_t) (I2S_MODE_SLAVE | I2S_MODE_RX)

There is no data on the data bus

**Reply by pschatzmann** [reply]
I never tried this, so this is not the right place to answer your question. I think you will need to analyze the signals to figure out what's going wrong.

The only thing I now for sure is that the Masterclock can only be output (=Master) on an ESP32 and never input.

You better use an ESP32 IDF support forum for this question.

**Reply by amirkakhki** [reply]
> I understand, but it didn't work for me, so I thought I'd ask, I did it like this:
> 
> .mode = (i2s_mode_t) (I2S_MODE_SLAVE | I2S_MODE_RX)
> 
> There is no data on the data bus

Hi, Could you fix this issue? I'm experiencing the same problem with I2S in slave mode.
i used this:
cfg.is_master=false;

**Reply by pschatzmann** [reply]
Why are you still using an old 2.x core version ? The actual version is 3.0.5 with 3.1.0 coming out soon.
Setting I2S to slave mode is with the AudioTools is

```C++
    auto config = i2s.defaultConfig(TX_MODE);
    config.sample_rate = sample_rate;
    config.channels = channels;
    config.bits_per_sample = 16;
    config.is_master = false;
    i2s.begin(config);
```

There is only data on the data bus when you are actually playing music. I don't recommend to these kind of tests with the A2DP library, but use e.g. [the sine test case of the audiotools](https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working).

ps. RX_MODE (or I2S_MODE_RX) would only be relevant when you use a ADC for the AudioSource scenario to send audio to an external Bluetooth speaker.

**Reply by Paul-TheOctopus** [reply]
Hi guys! I'm following in someone's footsteps. 
I can't get a signal from ESSP32 (in slave mode) to ADAU1701. I'm very confused by the presence of LR, BCLC signals on ESP32 in slave mode. In my opinion, they shouldn't be there.
@amirkakhki I'd be grateful if you share the results of your research and the problems you encountered.

**Reply by yentzee** [reply]
same here, i tried for 2 days and just couldnt get any data signal, i supplied  btck and lrck  from stm32 but no luck. changing to master now but the clocks are very jittery :(
